rfcs/min-relations.md

```diff
@@ -0,0 +1,270 @@
+# Feature Name: `min-relations`
+
+// FIXME
+// Make sure to talk about these:
+// - Reverse relations.
+// - Or support for add_target_filter.
+
+## Summary
+
+Relations connect entities to each other, storing data in these edges. This RFC covers the minimal version of this feature, allowing for basic entity grouping and simple game logic extensions.
+
+## Motivation
+
+Entities often need to be aware of each other in complex ways: an attack that targets another unit, groups of entities that are controlled en-masse, or complex parent-child hierarchies that need to move together.
+
+We *could* simply store an `Entity` in a component, then use `query.get(my_entity)` to access the relevant data.
+(In this RFC, we call this the **Entity-in-component pattern**.)
+But this quickly starts to balloon in complexity.
+
+- How do we ensure that these components are cleaned up when the entity they're pointing to is?
+- How do we handle pointing to multiple entities in similar ways?
+- How do we look for all entities that point to a particular entity of interest?
+- How do we despawn an entity and all of its children (with many types of child-like entities)?
+- How do we quickly access data on the entity that we're attached to?
+- How do we traverse this graph of entities?
+- How do we ensure that our graph is acylic?
```

Comment:
I'm not sure if this particular point is handled in any way by proposed relations. Afaict cycles are still an issue (in cases where you explicitly don't want them).

Reply:
The framing could be clearer. These problems are *possible* to address in a cohesive way at an engine-level by including relations as a feature. Min-relations won't hit all of these though :(

Reply:
Okay, so it's more of a future prospect. I guess that's fine.

Reply:
Clarified!